python/paddle/v2/framework/layers.py

```diff
@@ -324,31 +350,32 @@ def conv2d(input,
     return helper.append_activation(pre_act)
 
 
-def sequence_pool(input,
-                  pool_size,
-                  pool_type,
-                  pool_stride=1,
-                  pool_padding=0,
-                  global_pooling=False,
-                  program=None,
-                  init_program=None):
+def sequence_pool(input, pool_type, program=None, init_program=None):
     # FIXME(dzh) : want to unify the argument of python layer
     # function. So we ignore some unecessary attributes
 
-    ENUM_POOL_TYPE = set(["max", "avg", "sqrt", "last", "first"])
-    if pool_type not in ENUM_POOL_TYPE:
+    ENUM_POOL_TYPE = dict({
+        "AVERAGE": 0,
+        "SUM": 1,
+        "SQRT": 2,
+        "MAX": 3,
+        "LAST": 4,
+        "FIRST": 5
+    })
+    if pool_type.upper() not in ENUM_POOL_TYPE:
         raise ValueError("Unknown pool_type: '%s'. It can only be %s.",
-                         str(pool_type), " ".join(ENUM_POOL_TYPE))
+                         str(pool_type), " ".join(ENUM_POOL_TYPE.keys()))
 
     helper = LayerHelper('sequence_pool', **locals())
     dtype = helper.input_dtype()
     pool_out = helper.create_tmp_variable(dtype)
 
+    # FIXME(dzh): strategy
     helper.append_op(
         type="sequence_pool",
         inputs={"X": [input]},
-        outputs={"Out": pool_out},
-        attrs={"strategy": pool_type})
+        outputs={"Out": [pool_out]},
+        attrs={"strategy": ENUM_POOL_TYPE[pool_type.upper()]})
```

Comment:
What's this change for?

Reply:
"max", "sqrt" are keywords of `Python`, I cannot bear our keyword overloading anymore.....

Reply:
I am wondering why `ENUM_POOL_TYPE` becomes a set.

Reply:
Because attribute of "strategy" needs a fixed enum index, not a string. 
```
    AddAttr<int>(
        "strategy",
        "(int, default AVERAGE) the pooling strategy of SequencePoolOp.")
        .SetDefault(AVERAGE)
        .InEnum({AVERAGE, SUM, SQRT, MAX, LAST, FIRST});
```
And this attribute will be definitely replaced with another PR, and the sequence_conv attributes, etc. Those tons of code will be rewritten after I finished the book chapter 5.